integration/nwo/network.go

```diff
@@ -144,9 +144,11 @@ type Network struct {
 	StatsdEndpoint     string
 	ClientAuthRequired bool
 
-	PortsByBrokerID  map[string]Ports
-	PortsByOrdererID map[string]Ports
-	PortsByPeerID    map[string]Ports
+	PortsByBrokerID          map[string]Ports
+	PortsByOrdererID         map[string]Ports
+	PortsByPeerID            map[string]Ports
+	PortsByChaincodeServerID map[string]Ports
```

Comment:
The external chaincode is not part of the network proper; if you need to allocate a port to listen on, just call `network.ReservePort` to reserve the port for the chaincode instance(s) you need.

There's no need to change the model.